src/main/scala/org/hammerlab/guacamole/commands/GermlineAssemblyCaller.scala

```diff
@@ -289,49 +231,81 @@ object GermlineAssemblyCaller {
       DelayedMessages.default.print()
     }
 
-    def discoverGenotypes(reads: RDD[MappedRead],
+    def discoverGermlineVariants(reads: RDD[MappedRead],
                           kmerSize: Int,
                           snvWindowRange: Int,
                           minOccurrence: Int,
                           minAreaVaf: Float,
                           reference: ReferenceBroadcast,
-                          lociPartitions: LociMap[Long]): RDD[CalledAllele] = {
+                          lociPartitions: LociMap[Long],
+                          minAltReads: Int = 2,
+                          minPhredScaledLikelihood: Int = 0,
+                          shortcutAssembly: Boolean = false): RDD[CalledAllele] = {
 
       val genotypes: RDD[CalledAllele] =
-        windowFlatMapWithState[MappedRead, CalledAllele, Option[DeBruijnGraph]](
+        windowFlatMapWithState[MappedRead, CalledAllele, Option[Long]](
           Vector(reads),
           lociPartitions,
           skipEmpty = true,
           halfWindowSize = snvWindowRange,
           initialState = None,
-          (graph, windows) => {
+          (lastCalledLocus, windows) => {
             val window = windows.head
+            val referenceName = window.referenceName
+            val locus = window.currentLocus
+
+            val referenceStart = (locus - window.halfWindowSize).toInt
+            val referenceEnd = (locus + window.halfWindowSize).toInt
+            val currentReference =
+              reference.getReferenceSequence(
+                window.referenceName,
+                referenceStart,
+                referenceEnd
+              )
+
             // Find the reads the overlap the center locus/ current locus
             val currentLocusReads =
               window
                 .currentRegions()
                 .filter(_.overlapsLocus(window.currentLocus))
-            val variableReads =
-              currentLocusReads
-                .count(read =>
-                  read.cigar.numCigarElements() > 1 || read.countOfMismatches(reference.getContig(window.referenceName)) > 0)
+
+            val pileup = Pileup(currentLocusReads, referenceName, window.currentLocus, reference.getContig(referenceName))
 
             // Compute the number reads with variant bases from the reads overlapping the currentLocus
-            val currentLocusVAF = variableReads.toFloat / currentLocusReads.length
-            if (currentLocusVAF > minAreaVaf) {
-              val result = discoverHaplotypes(
+            val pileupAltReads = (pileup.depth - pileup.referenceDepth)
+            if (currentLocusReads.isEmpty || pileupAltReads < minAltReads) {
+              (lastCalledLocus, Iterator.empty)
+            } else if (shortcutAssembly &&
+              currentLocusReads.count(r => r.countOfMismatches > 1 || r.cigar.numCigarElements() > 1) / currentLocusReads.size.toFloat < minAreaVaf) {
```

Comment:
Yes, good catch, I had these named differently before and somehow mixed them up.